Automattic/themes
# Move Blockbase Google Fonts to Locally Hosted ( DO NOT MERGE )

#### Changes proposed in this Pull Request:

- Migrate custom font settings to be more compatible with Global Styles (solves https://github.com/Automattic/wp-calypso/issues/63229)
- Add Google Fonts to theme.json to populate Font family options in the editor
- Include Jetpack package for loading Google fonts from the editor settings, so that Jetpack plugin is not required
- Remove Customizer font settings and add deprecation message that links to Global styles
- Disable Jetpack Google Fonts module, if active, so it doesn't conflict with Blockbase fonts

**Note:** This migrates the Heading font from the Customizer to the Global Styles settings for the Post Title and Heading blocks. The Heading block typography settings require Gutenberg 13.3, which has not yet been released, so this PR should wait to be merged until then.

#### Testing Instructions
- Activate a blockbase theme ( Quadrat, Bennett, Blockbase, etc. )
- Before applying this branch, set custom fonts from Customizer > Fonts

<img width="1496" alt="Screen Shot 2022-05-25 at 12 21 05 AM" src="https://user-images.githubusercontent.com/5414230/170204672-381076fe-7f55-430f-a0b0-73ac65132d6b.png">

- Navigate to Appearance > Editor ( beta )
- Verify that custom google fonts are not selectable in the global styles typography dropdown of the site editor

<img width="1496" alt="Screen Shot 2022-05-25 at 12 22 56 AM" src="https://user-images.githubusercontent.com/5414230/170204931-838e5f74-1758-4be4-b8a5-6c58ae52c89c.png">

- Apply this branch and reload a page to migrate the settings
- You should see the body font you selected in Global Styles > Typography

<img width="1496" alt="Screen Shot 2022-05-25 at 12 26 46 AM" src="https://user-images.githubusercontent.com/5414230/170205647-f8f78b30-92d8-4db2-a700-b56b29ceafef.png">

- You should see the heading font you selected in Global Styles > Blocks > Heading > Typography (currently requires Gutenberg `trunk`) and Global Styles > Blocks > Post Title > Typography

<img width="1496" alt="Screen Shot 2022-05-25 at 12 26 58 AM" src="https://user-images.githubusercontent.com/5414230/170205634-9c4877ba-acfd-4656-891d-deaf10f8b230.png">

- You should see all the fonts that Blockbase provides in the list of font families provided by Global Styles and Block Settings

<img width="1496" alt="Screen Shot 2022-05-25 at 12 27 58 AM" src="https://user-images.githubusercontent.com/5414230/170205683-cfca11fa-b32e-45a0-8f49-7ae152d51db3.png">

- When you navigate back to the customizer, you should see a deprecation notice
<img width="1096" alt="Screen Shot 2022-05-24 at 11 49 09 PM" src="https://user-images.githubusercontent.com/5414230/170198960-f7657e16-8024-4043-8c42-2f642f582ffc.png">

**Note:**
- To test repeatedly, delete the `blockbase_legacy_font_settings` from wp_options database table to trigger the settings migration (migration only runs when this option is not present).
